### PR TITLE
[source-mssql/mysql/postgres] Fix and cleanup oc map

### PR DESCRIPTION
**Pull request by xiaohansong**
## What

https://github.com/airbytehq/oncall/issues/5825

We need to pre-populate all ocInfo or pkInfo to state manager. 

In case like this:

We have 2 resumable full refreshes. It failed when we are syncing the 1st one. The final state would be like 
{stream: A, streamState: { ...(detailed state)... } }
{stream: B, streamState: null }

The next time it load, with previous logic, it will only load for pkInfo(ocInfo) for stream A. When we run load records for stream B, pkInfo or ocInfo would be null. That is confirmed by the added test.

## How
Added a test to cover this. There are some cdk traffic about error message and wass interface change - I just stubbed them. will wait them to merge before checking this in.

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/exceptions/ConfigErrorException.kt [hunk 1]
```diff
@@ -11,6 +11,10 @@ class ConfigErrorException : RuntimeException {
     val internalMessage: String
     val displayMessage: String
 
+    constructor(displayMessage: String) : super(displayMessage) {
```
Comment (xiaohansong) [inline airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/exceptions/ConfigErrorException.kt:14]: will revert. Adding this so I can run some tests.

**Comment by theyueli** [comment]
@xiaohansong , also fixes this https://github.com/airbytehq/airbyte-internal-issues/issues/8560 ?

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadStateManager.java [hunk 2]
```diff
@@ -41,7 +41,13 @@ public void updatePrimaryKeyLoadState(final AirbyteStreamNameNamespacePair pair,
 
   // Returns the previous state emitted, represented as a {@link PrimaryKeyLoadStatus} associated with
   // the stream.
-  public abstract PrimaryKeyLoadStatus getPrimaryKeyLoadStatus(final AirbyteStreamNameNamespacePair pair);
+  public PrimaryKeyLoadStatus getPrimaryKeyLoadStatus(final AirbyteStreamNameNamespacePair pair) {
+    if (pairToPrimaryKeyLoadStatus.containsKey(pair)) {
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadStateManager.java:45]: simple doing `get()` does the same thing, no?

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialReadUtil.java [hunk 3]
```diff
@@ -159,7 +159,7 @@ public static boolean isSavedOffsetStillPresentOnServer(final JdbcDatabase datab
       AirbyteTraceMessageUtility.emitAnalyticsTrace(cdcCursorInvalidMessage());
       if (!sourceConfig.get("replication_method").has(INVALID_CDC_CURSOR_POSITION_PROPERTY) || sourceConfig.get("replication_method").get(
           INVALID_CDC_CURSOR_POSITION_PROPERTY).asText().equals(FAIL_SYNC_OPTION)) {
-        throw new ConfigErrorException(
+        throw new ConfigErrorException("test",
```
[the remark this answers is not in the document]
Comment (xiaohansong) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialReadUtil.java:162]: ah yeah just to bypass the compile error when I draft this PR. I'll revert this!

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadStateManager.java (hunk 2 above)
Comment (xiaohansong) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadStateManager.java:45]: yeah you are right... I was paranoid that this was null

Comment (xiaohansong) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadStateManager.java]: fixed!

**Review by rodireich — APPROVED** [review]

**Pull request merged**